geniza/corpus/management/commands/add_fragment_urls.py

```diff
@@ -0,0 +1,144 @@
+import csv
+import re
+from collections import defaultdict
+
+from django.conf import settings
+from django.contrib.admin.models import CHANGE, LogEntry
+from django.contrib.auth.models import User
+from django.contrib.contenttypes.models import ContentType
+from django.core.management.base import BaseCommand, CommandError
+from parasolr.django.signals import IndexableSignalHandler
+
+from geniza.corpus.models import Fragment
+
+
+class Command(BaseCommand):
+    """Takes a CSV of shelfmarks and view URLs and/or IIIF URLs, update
+    corresponding Fragment records in the database with those URLs.
+    Expects CSV headers 'shelfmark' and one or both of 'url' and 'iiif_url'"""
+
+    help = __doc__
+
+    def __init__(self, *args, **options):
+        super().__init__(*args, **options)
+
+        self.stats = defaultdict(int)
+
+        self.fragment_contenttype = ContentType.objects.get_for_model(Fragment)
+        self.script_user = User.objects.get(username=settings.SCRIPT_USERNAME)
+
+        # disconnect solr indexing signals
+        IndexableSignalHandler.disconnect()
+
+    def add_arguments(self, parser):
+        parser.add_argument("csv", type=str)
+        parser.add_argument("-o", "--overwrite", action="store_true")
+        parser.add_argument("-d", "--dryrun", action="store_true")
+
+    def handle(self, *args, **options):
+        self.csv_path = options.get("csv")
+        self.overwrite = options.get("overwrite")
+        self.dryrun = options.get("dryrun")
+
+        try:
+            with open(self.csv_path) as f:
+                csvreader = csv.DictReader(f)
+                for row in csvreader:
+                    if "shelfmark" not in row or not (
+                        "url" in row or "iiif_url" in row
+                    ):
+                        raise CommandError(
+                            "CSV must include 'shelfmark' and one or both of 'url' and 'iiif_url'"
+                        )
+                    self.add_fragment_urls(row)
+        except FileNotFoundError as err:
+            raise CommandError(err)
```

Comment:
fwiw when i tested this pattern on `merge_joins`, the `FileNotFoundError` doesn't actually report the path, so the error text is kinda useless. I just raised `CommandError(f"csv file not found: {self.csv_path}")` or something like that to make sure that it was displayed